ex04.ex

```diff
@@ -39,7 +39,9 @@ defmodule Ex04 do
       [ 1, 2, 3, 4, 5 ]
 
   """
-  def reverse . . . "your code"
+
+  def reverse([ ]), do: reduce([ ], [ ], &[ &1 | &2 ])
```

Comment:
# -2

There are two things wrong with this line:

~~~ elixir
def reverse([ ]), do: reduce([ ], [ ], &[ &1 | &2 ])
~~~

First, the reverse of an empty list is an empty list, so it should just be

~~~ elixir
def reverse([ ]), do: []
~~~

Second, it isn't needed at all, because `reduce` already handles that case.